src/pages/payment/Payment.tsx

```diff
@@ -0,0 +1,78 @@
+import { useParams } from 'react-router-dom';
+import { Elements } from '@stripe/react-stripe-js';
+import { ChangeEvent, FormEvent, useCallback, useEffect, useState } from 'react';
+import { Box, FormControl, FormControlLabel, FormLabel, Radio, RadioGroup } from '@mui/material';
+import { PAYMENT_OPTION } from './components/contants';
+import { PaymentForm } from './components/PaymentForm';
+import { stripePromise } from '../../config/stripe.config';
+import ButtonPrimary from '../../components/button/ButtonPrimary';
+import LoadingPrimary from '../../components/loader/LoadingPrimary';
+import usePostPaymentOptionMutation from '../../api/mutations/payment/usePostPaymentOptionMutation';
+import DataFetchError from '../../components/shared/DataFetchError';
+
+const Payment = () => {
+  const bookingId = useParams().id;
+  const [paymentOption, setPaymentOption] = useState<string>(PAYMENT_OPTION.card);
+
+  const { mutate, isPending, isError } = usePostPaymentOptionMutation(
+    bookingId || '',
+    paymentOption
+  );
+
+  useEffect(() => {
+    mutate();
```

Comment:
Why do you need mutation in `useEffect`

Reply:
In the first render, I should need to send a request